src/stores/RoomScrollStateStore.js

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Stores where the user has scrolled to in each room
+ */
+class RoomScrollStateStore {
+    constructor() {
+        this._scrollStateMap = {};
+    }
+
+    getScrollState(roomId) {
+        return this._scrollStateMap[roomId];
+    }
+
+    setScrollState(roomId, scrollState) {
+        this._scrollStateMap[roomId] = scrollState;
+    }
+}
+
+if (global.mx_RoomScrollStateStore === undefined) {
+    global.mx_RoomScrollStateStore = new RoomScrollStateStore();
```

Comment:
I'm not entirely convinced we want to be sticking more stuff in the global namespace?

Reply:
Unfortunately I still don't know of any other way to make a proper singleton (declaring them as module level variables gives you a different instance when accessed from a different module which causes subtle breakage when accessing between riot-web and react-sdk).

Reply:
https://derickbailey.com/2016/03/09/creating-a-true-singleton-in-node-js-with-es6-symbols/ suggests alternatives which are complicated. fair enough.